### PR TITLE
Add lpla ddi for hevc

### PR DESCRIPTION
**Pull request by Xuhenry1**
lpla ddi added for hevc.

### CODE REVIEW

File: va/va.h [hunk 1]
```diff
@@ -3770,6 +3787,27 @@ VAStatus vaBufferSetNumElements (
  */
 #define VA_CODED_BUF_STATUS_SINGLE_NALU                 0x10000000	
 
+typedef struct _VALookAheadInfo
+{
+    uint32_t status_report_feedback_number;
```
Comment (XinfengZhang) [inline va/va.h:3792]: please add description for each variable

File: va/va.h [hunk 2]
```diff
@@ -3799,8 +3837,10 @@ typedef  struct _VACodedBufferSegment  {
      */
     void               *next;
 
+    VALookAheadInfo   lpla_info;
+
     /** \brief Reserved bytes for future use, must be zero */
-    uint32_t                va_reserved[VA_PADDING_LOW];
+    //uint32_t                va_reserved[VA_PADDING_LOW];
```
Comment (XinfengZhang) [inline va/va.h:3843]: the structure of VALookAheadInfo is five DWORD, so , this commit change the size of this structure.

**Review by XinfengZhang — CHANGES_REQUESTED** [review]

File: va/va.h [hunk 3]
```diff
@@ -3770,6 +3785,26 @@ VAStatus vaBufferSetNumElements (
...
+typedef struct _VALookAheadInfo
+{
+    union
+    {
+        struct
+        {
+            uint32_t valid_info   : 1;
+            uint32_t cqm_hint     : 8;        // Custom quantization matrix hint. 0x00 - flat matrix; 0x01 - CQM; 0xFF - invalid hint; other values are reserved.
+            uint32_t intra_hint   : 1;
+            uint32_t min_gop_size : 8;        // [0, 1, 2, 4, 8,16]
```
Comment (cchunbo) [inline va/va.h:3797]: why put this min_gop_size inside the encoderHints union?

File: va/va.h (hunk 2 above)
Comment (cchunbo) [reply va/va.h:3840]: same question, is the VALookAheadInfo size exactly the same as va_reserved?

File: va/va_enc_hevc.h [hunk 4]
```diff
@@ -338,8 +339,26 @@ typedef struct _VAEncSequenceParameterBufferHEVC {
         } bits;
         uint32_t value;
     } scc_fields;
-    /** \brief Reserved bytes for future use, must be zero */
-    uint32_t   va_reserved[VA_PADDING_MEDIUM - 1];
+
+    /** \brief Lookahead depth in frames. */
+    uint8_t     lookahead_depth;
+
+    uint8_t     min_adaptive_gop_pic_size;
```
Comment (cchunbo) [inline va/va_enc_hevc.h:346]: is below changes related with lookahead? how is it used?

File: va/va_enc_hevc.h [hunk 5]
```diff
@@ -560,8 +579,20 @@ typedef struct _VAEncPictureParameterBufferHEVC {
...
+    {
+        struct
+        {
+            uint8_t    x16_minus1_x : 4; // scaling ratio = (X16Minus1_X + 1) / 16
+            uint8_t    x16_minus1_y : 4;
+        } bits;
+        uint8_t    value;
+    } downscale_ratio;
+
+    uint8_t qp_modulation_strength;
```
Comment (cchunbo) [inline va/va_enc_hevc.h:593]: where is the target frame size?

**Pull request closed without merging**